src/chopper_layout.cpp

```diff
@@ -290,6 +294,19 @@ int execute(seqan3::argument_parser & parser)
 
     // brief Write the output to the result file.
     std::ofstream fout{config.output_filename};
+
+    // write the config
+    fout << "##CONFIG:\n";
+    std::stringstream config_stream{};
+    cereal::JSONOutputArchive output(config_stream); // stream to cout
+    output(cereal::make_nvp("config", config));
+    std::string line;
+    while (std::getline(config_stream, line, '\n'))
+        fout << "##" << line << '\n';
+    fout << "##}\n" // last closing bracket isn't written by loop above
+         << "##ENDCONFIG\n";
```

Comment:
I think I would still prefer some other prefix (`@`?).
Makes parsing downstream easier, because you always just need to check the first character.

Reply:
Hm, I think it is easier if the very first character is the same. That way you have a clear specification of what line belongs to the header in case you want to skip it while parsing. I also saw in Hannes IO library that for plain text IO (which would be usable here) you can specifiy "header starts with:". Which only allows one character though.

After the `#` we could then add `@` if you want?

Reply:
Mhm, that's also a good point.
Then we just leave it at

```
## config
## ...
# actual header
# ...
```

Reply:
However, I wonder if it would make sense to refactor writing the config into its own function?
```cpp
void write_config_to(configuration const & config, std::ofstream & stream)
{
    // ...
}
```

Reply:
yes, good point